### DISCUSSION TITLE
any way to ignore "=" present in CSV

### DISCUSSION THREAD

**Discussion by focusede** [JoshClose/CsvHelper#1461] [Q&A]
Hey all

I have used CSVhelper a bunch of times and love it.  I have some applications where they are "helpfully" including the excel fix of "=" as a preamble to quoted 0 filled numerics to prevent the Excel interpreter from eating the leading 0.  CSVhelper reports this row as an error.  Other than search/replace to pull the "=" out, is there a way to tell CSVhelper to ignore this equals and process successfully?  I see options for including = in the written output but not to allow them in the parse.  Thanks

**Comment by JoshClose** [comment]
All Excel functionality was removed, except for injection sanitization. You should be able to do it fairly easily by creating your own parser.

```cs
void Main()
{
	var s = new StringBuilder();
	s.AppendLine("Id,Name");
	s.AppendLine("=01,one");
	s.AppendLine("=02,two");
	using (var reader = new StringReader(s.ToString()))
	using (var parser = new ExcelParser(reader, CultureInfo.InvariantCulture))
	using (var csv = new CsvReader(parser))
	{
		csv.GetRecords<Foo>().ToList().Dump();
	}
}

public class Foo
{
	public int Id { get; set; }
	public string Name { get; set; }
}

public class ExcelParser : CsvParser
{
	public ExcelParser(TextReader textReader, CultureInfo culture) : base(textReader, culture) { }

	public override string[] Read()
	{
		var row = base.Read();
		if (row == null)
		{
			return row;
		}
		
		for (var i = 0; i < row.Length; i++)
		{
			row[i] = row[i].TrimStart('=');
		}
		
		return row;
	}
}
```

**Reply by icecactus** [reply]
Is there an alternate way of doing this? You can no longer override the Read method.

**Reply by icecactus** [reply]
Im posting an update to this, so anyone in the future that runs across this hopefully doesnt spend way to much time like I did looking into parsers etc The solution below works great and is super simple. 

See https://github.com/JoshClose/CsvHelper/issues/1879

var config = new CsvConfiguration(CultureInfo.InvariantCulture)
{
    TrimOptions = TrimOptions.Trim,
    WhiteSpaceChars = new char[] { '=' },
};


Perhaps you should change WhiteSpaceChars to TrimChars. I looked into trim options, but didnt realize you could specify the characters. Naming it TrimChars, I think I would have realized I could specify what characters to trim in intellisense :)
